serial/components/virt_rx.c

```diff
@@ -174,7 +178,11 @@ void handle_rx() {
                 } else {
                     // Ensure that multi client input is off
                     multi_client = 0;
-                    int new_client = atoi(&got_char);
+                    // Null terminate got_char to be safe to use with atoi
+                    char got_char_terminated[2];
+                    got_char_terminated[0] = got_char;
+                    got_char_terminated[1] = '\0';
+                    int new_client = atoi(got_char_terminated);
```

Comment:
`atoi()` is for converting strings, not single characters, please just use:

```c
int new_client = chars[0] - '0';
```

Reply:
That won't work for more than 10 clients though, but neither does the current code.

All of this code will be revisited soon anyways with https://github.com/au-ts/sddf/issues/53. I've added a note about the use of `atoi` in the issue.